### PR TITLE
Fixes response on filenames with weird characters

### LINKED ISSUE #16: Upload fails, still get back the file after that using GET /records
Files that not successfully uploaded should be filtered out

### PR DESCRIPTION
**Pull request by blootsvoets**
Improve handling of filenames with symbols. Fixes #16 and #17.

### CODE REVIEW

File: radar-upload-backend/src/main/java/org/radarbase/upload/api/RecordMapperImpl.kt [hunk 1]
```diff
@@ -73,12 +74,16 @@ class RecordMapperImpl : RecordMapper {
...
 
-    override fun fromContent(content: RecordContent) = ContentsDTO(
-            url = "${uri.baseUri}/records/${content.record.id}/contents/${content.fileName}",
-            contentType = content.contentType,
-            createdDate = content.createdDate,
-            size = content.size,
-            fileName = content.fileName)
+    override fun fromContent(content: RecordContent): ContentsDTO {
+        val cleanedFileName = URLEncoder.encode(content.fileName, "UTF-8")
+                .replace("+", "%20")
```
Comment (nivemaham) [inline radar-upload-backend/src/main/java/org/radarbase/upload/api/RecordMapperImpl.kt]: Why do we replace only this character after encoding? :thinking:

Comment (blootsvoets) [reply radar-upload-backend/src/main/java/org/radarbase/upload/api/RecordMapperImpl.kt]: It's based on this answer: https://stackoverflow.com/a/31722987/574082. Apparently the encoder works fine, except that it translates spaces to +.

Comment (blootsvoets) [reply radar-upload-backend/src/main/java/org/radarbase/upload/api/RecordMapperImpl.kt]: Added a test to ensure that the URL is correct with symbols.

Comment (nivemaham) [reply radar-upload-backend/src/main/java/org/radarbase/upload/api/RecordMapperImpl.kt]: ok. Thanks for the info

File: radar-upload-backend/build.gradle.kts [hunk 2]
```diff
@@ -67,6 +67,7 @@ dependencies {
     testImplementation("org.junit.jupiter:junit-jupiter:5.4.2")
     testImplementation("org.mockito:mockito-core:2.21.0")
     testImplementation("org.hamcrest:hamcrest-all:1.3")
+    testImplementation("com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0")
```
Comment (nivemaham) [inline radar-upload-backend/build.gradle.kts]: we already have mockito in our dependencies.Maybe it is good to move all of the tests with mockitokotlin

**Pull request merged**